### ISSUE TITLE
Property `nullable: false` attribute not respected

### ISSUE THREAD

**Issue by micchickenburger** [mikro-orm/mikro-orm#3419]
**Describe the bug**
I have a property that I want the database to auto populate at insertion.  Therefore, I have the property set as optional in the class so that it is not required when instantiating.  But the database isn't populating the field with the default value because null values are allowed despite setting `nullable` to false in the property declaration.  I'm using ts-morph.

It seems only `nullable: true` is respected:

```typescript
export abstract class Test {
  @Property() // "test1" timestamptz(0) not null
    test1!: Date;

  @Property() // "test2" timestamptz(0) null
    test2?: Date;

  @Property({ nullable: true }) // "test3" timestamptz(0) null
    test3!: Date;

  @Property({ nullable: true }) // "test4" timestamptz(0) null
    test4?: Date;

  @Property({ nullable: false }) // "test5" timestamptz(0) not null
    test5!: Date;

  @Property({ nullable: false }) // "test6" timestamptz(0) null  < seems incorrect
    test6?: Date;
}
```

**Expected behavior**
I would expect `nullable: false` to take precedence over the property signature.

**Versions**

| Dependency | Version |
| - | - |
| node | v16.16.0 |
| typescript | 4.7.4 |
| mikro-orm | 5.2.4 |
| postgresql | 5.2.4 |

**Comment by Langstra** [comment]
In my opinion this is not a bug, but more like undefined behaviour. 

Maybe that the ORM should show you an error/warning when you try to create a optional property with conflicting annotation?

**Comment by micchickenburger** [comment]
Thanks for your reply, @Langstra.  The trouble is that if the property is required a value has to be provided, and if the value is provided it won't be auto populated in the database.  In particular, I am looking for a timestamp that reflects the insertion time, not the time during processing, which the application would provide.

Do you have an example pattern that could make this work?

**Comment by Langstra** [comment]
You could use the `defaultRaw` property of the `Property` decorator to call `now`.

https://mikro-orm.io/docs/4.5/defining-entities#default-values

**Comment by B4nan** [comment]
This seems valid to me, there is no reason to override user provided values. PR welcome, I dont have time for this before next week I guess.

**Comment by micchickenburger** [comment]
> You could use the `defaultRaw` property of the `Property` decorator to call `now`.
> 
> https://mikro-orm.io/docs/defining-entities#default-values

This is the approach that I'm using.  The trouble is that required values are required during class instantiation.  However, for `defaultRaw` to work, the value cannot be passed during the SQL insert, correct?  So for a default value to be populated the property has to be marked as optional.  Maybe I'm missing something obvious.

**Comment by B4nan** [comment]
Fixed via [75a6487](https://github.com/mikro-orm/mikro-orm/commit/75a6487cc45eefd5ca7e4bd530b0522222e632ee), please try the dev version (will be out in few minutes once the CI finishes).
